torch/_dynamo/variables/user_defined.py

```diff
@@ -204,11 +205,12 @@ class UserDefinedObjectVariable(UserDefinedVariable):
     Mostly objects of defined type.  Catch-all for something where we only know the type.
     """
 
-    def __init__(self, value, value_type=None, **kwargs):
+    def __init__(self, value, value_type=None, torch_function_fn=None, **kwargs):
         super().__init__(**kwargs)
         self.value = value
         self.value_type = value_type or type(value)
         assert type(value) is self.value_type
+        self.torch_function_fn = torch_function_fn
```

Comment:
@jansel @voznesenskym so there's this pattern of passing in "inner" variable trackers during construction of variable trackers and it feels like this is counter to the upcoming lazy VT regime. But I am not sure what the correct style here is. Should the constructor here recursively call builder? Should this be forced to be a non-optional argument?

Reply:
I originally had a separate variable tracker inheriting from UDOV, but Ed suggested just adding functionality to UDOV because (I think) it's more representative of the python object model. Curious what you guys think

Reply:
Shouldn't this be a method on UserDefinedObjectVariable as opposed to a property?

We should only check (and guard on) torch_function if the object is used in a place where having a torch function matters.  Doing it outside the constructor seems odd.